File: src/javascript/algorithms/binary/reverseBits.js
// TC:O(32) ~ O(1) SC:O(1)

function reverseBits1(num) {
    let reverseBitResult = 0;
    for(let i=0; i< 32; i++) {
        reverseBitResult = reverseBitResult << 1;
        reverseBitResult = reverseBitResult | (1 & num);
        num = num >> 1;
    }
    return reverseBitResult;
}

function reverseBits2(num) {
    let reverseBitResult = 0;
    for(let i=0; i<32; i++) {
        if((num >> i & 1) === 1)
        reverseBitResult = reverseBitResult | 1 << 31-i;
    }
    return reverseBitResult;
}

let num = 11;
console.log(reverseBits1(num));
console.log(reverseBits2(num));

File: src/java1/algorithms/binary/ReverseBits.java
// TC:O(32) ~ O(1) SC:O(1)
package java1.algorithms.binary;

public class ReverseBits {
    private static int reverseBits1(int n) {
        int reverseBits = 0;
        for(int i=0; i< 32; i++) {
            reverseBits = reverseBits << 1;
            reverseBits = reverseBits | (1 & n);
            n = n >> 1;
        }
        return reverseBits;
    }

    private static int reverseBits2(int n) {
        int reverseBits = 0;
        for(int i=0; i< 32; i++) {
            if((n >> i & 1) == 1)
            reverseBits = reverseBits | (1 << (31 - i));
        }
        return reverseBits;
    }

    public static void main(String[] args) {
        int num = 11;
        System.out.println(reverseBits1(num));
        System.out.println(reverseBits2(num));
    }
}
